cpp/velox/udf/UdfLoader.cc

```diff
@@ -77,11 +78,34 @@ std::unordered_set<std::shared_ptr<UdfLoader::UdfSignature>> UdfLoader::getRegis
       const auto& entry = udfEntry[i];
       auto dataType = toSubstraitTypeStr(entry.dataType);
       auto argTypes = toSubstraitTypeStr(entry.numArgs, entry.argTypes);
-      signatures.insert(std::make_shared<UdfSignature>(entry.name, dataType, argTypes));
+      if (entry.intermediateType) {
+        auto intermediateType = toSubstraitTypeStr(entry.intermediateType);
+        signatures_.insert(std::make_shared<UdfSignature>(entry.name, dataType, argTypes, intermediateType));
+      } else {
+        signatures_.insert(std::make_shared<UdfSignature>(entry.name, dataType, argTypes));
+      }
     }
     free(udfEntry);
   }
-  return signatures;
+  return signatures_;
+}
+
+std::unordered_set<std::string> UdfLoader::getRegisteredUdafNames() {
+  if (handles_.empty()) {
+    return {};
+  }
+  if (!names_.empty()) {
+    return names_;
+  }
+  if (signatures_.empty()) {
+    getRegisteredUdfSignatures();
+  }
+  for (const auto& sig : signatures_) {
+    if (!sig->intermediateType.empty()) {
```

Comment:
Maybe, better to not mix UDF/UDAF signatures together?

Reply:
Thanks. I've also considered this design. Will address that.